src/backend/joanie/tests/core/test_api_course.py

```diff
@@ -0,0 +1,317 @@
+
+        self.assertEqual(response.status_code, 401)
+        self.assertFalse(models.Course.objects.exists())
+
+    def test_api_course_create_authenticated(self):
+        """
+        Authenticated users should not be able to create a course.
+        """
+        user = factories.UserFactory(
+            is_staff=random.choice([True, False]),
+            is_superuser=random.choice([True, False]),
+        )
+        token = self.get_user_token(user.username)
+
+        data = {
+            "code": "ORG-001",
+            "title": "Organization 001",
+        }
+
+        response = self.client.post(
+            "/api/v1.0/courses/",
+            data=data,
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 405)
+        self.assertFalse(models.Course.objects.exists())
+
+    def test_api_course_update_anonymous(self):
+        """
+        Anonymous users should not be able to update a course.
+        """
+        course = factories.CourseFactory()
+
+        data = {
+            "code": "ORG-001",
+            "title": "Organization 001",
+        }
+
+        response = self.client.put(f"/api/v1.0/courses/{course.id}/", data=data)
+
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(
+            response.json(), {"detail": "Authentication credentials were not provided."}
+        )
+
+        course.refresh_from_db()
+        for key, value in data.items():
+            self.assertNotEqual(value, getattr(course, key))
+
+    def test_api_course_update_authenticated(self):
+        """
+        Authenticated users should be able to update a course.
+        """
+        user = factories.UserFactory(
+            is_staff=random.choice([True, False]),
+            is_superuser=random.choice([True, False]),
+        )
+        token = self.get_user_token(user.username)
+
+        course = factories.CourseFactory()
+
+        data = {
+            "code": "ORG-001",
+            "title": "Organization 001",
+        }
+
+        response = self.client.put(
+            f"/api/v1.0/courses/{course.id}/",
+            data=data,
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 405)
+        self.assertEqual(response.json(), {"detail": 'Method "PUT" not allowed.'})
+
+        course.refresh_from_db()
+        for key, value in data.items():
+            self.assertNotEqual(value, getattr(course, key))
+
+    def test_api_course_delete_anonymous(self):
+        """
+        Anonymous users should not be able to delete a course.
+        """
+        course = factories.CourseFactory()
+
+        response = self.client.delete(f"/api/v1.0/courses/{course.id}/")
+
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(models.Course.objects.count(), 1)
+
+    def test_api_course_delete_authenticated(self):
```

Comment:
in all these tests, aren't you forgetting to test for users who have an access to courses? I can be owner, administrator, instructor or manager... and maybe some of these roles do have the right to get/update/delete?

Reply:
I would say you need to make sure it returns a 405 because someone could allow it unintentionally.

Reply:
I added tests to ensure that `update/delete` does not work with owner user.